File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxrngrn.java
// EXPECTED: 70.00
// Reference emit for IDXRNGRN — START-from-key range sum (VSAM START substitute).
public final class IdxrngrnRef {
  public static void main(String[] args) {
    java.util.Map<Integer, Double> rows =
        java.util.Map.of(10, 12.50, 20, 8.00, 42, 25.00, 55, 30.00, 99, 15.00);
    double total = 0;
    for (var e : rows.entrySet()) {
      if (e.getKey() >= 42) total += e.getValue();
    }
    System.out.printf("%.2f%n", total);
  }
}

File: fixtures/hub-cobol-clbs-mini/batch/reference_emit_idxrngrn.py
"""Reference emit for IDXRNGRN — START-from-key range sum → 70.00."""
# EXPECTED: 70.00
ROWS = {10: 12.50, 20: 8.00, 42: 25.00, 55: 30.00, 99: 15.00}
START = 42
total = sum(v for k, v in ROWS.items() if int(k) >= START)
print(f"{total:.2f}")
